portia/runner.py

```diff
@@ -79,19 +79,26 @@ def __init__(
             tools (ToolRegistry | list[Tool]): The registry or list of tools to use.
 
         """
-        logger_manager.configure_from_config(config)
-        self.config = config
-        self.tool_registry = (
-            InMemoryToolRegistry.from_local_tools(tools) if isinstance(tools, list) else tools
-        )
+        self.config = config if config else Config.from_default()
+        logger_manager.configure_from_config(self.config)
+
+        if tools is None:
+            if self.config.portia_api_key:
+                self.tool_registry = PortiaToolRegistry.default(self.config)
+            else:
+                self.tool_registry = open_source_tool_registry
```

Comment:
Might be worth getting @mounir-portia view but I think we probably should alter slightly to:

```suggestion
            self.tool_registry = open_source_tool_registry
            if self.config.portia_api_key:
                self.tool_registry += PortiaToolRegistry.default(self.config)

        
            
```

Reply:
Might also be worth pulling this logic out into a function just to reduce the complexity in this init func?